.code-samples.meilisearch.yaml

```diff
@@ -0,0 +1,494 @@
+# This code-samples file is used by the MeiliSearch documentation
+# Every example written here will be automatically fetched by
+# the documentation on build
+# You can read more on https://github.com/meilisearch/documentation/tree/master/.vuepress/code-samples
+---
+get_one_index_1: |-
+  let movie_index: Index = client.get_index("movies").await.unwrap();
+list_all_indexes_1: |-
+  let indexes: Vec<Index> = client.list_all_indexes().await.unwrap();
+create_an_index_1: |-
+  let movie_index: Index = client.create_index("movies", Some("movie_id")).await.unwrap();
+update_an_index_1: |-
+  let movie_review_index: Index = client.get_index("movie_review").await.unwrap();
+  movie_review_index.update("movie_review_id").await.unwrap();
+delete_an_index_1: |-
+  movie_index.delete().await.unwrap();
+get_one_document_1: |-
+  let movie: Movie = movie_index.get_document(String::from("25684")).await.unwrap();
+get_documents_1: |-
+  let movie: Vec<Movie> = movie_index.get_documents(None, Some(2), None).await.unwrap();
+add_or_replace_documents_1: |-
+  // Note that the data differs a bit from other examples because we are using the exact same format as the database you downloaded, and the other examples are not
+  let progress: Progress = movie_index.add_or_replace(&[
+    Movie {
+      id: "287947".to_string(),
+      title: "Shazam".to_string(),
+      poster: "https://image.tmdb.org/t/p/w1280/xnopI5Xtky18MPhK40cZAGAOVeV.jpg".to_string(),
+      overview: "A boy is given the ability to become an adult superhero in times of need with a single magic word.".to_string(),
+      release_date: 1553299200,
+      genre: vec![
+        "Action".to_string(),
+        "Comedy".to_string(),
+        "Fantasy".to_string()
+      ]
```

Comment:
We try to follow the context in which the code samples are showcased.
In this case it is [add or replace documents](https://docs.meilisearch.com/references/documents.html#add-or-replace-documents) from which their is a provided body above the code samples.

This body does not have the `genre` field.
```suggestion
```

Reply:
I understand you wanting to use the movie dataset for every code sample. But the idea in the documentation is to keep every example as small as possible.  By adding those unecessary lines it clutters the examples and make them less readable. 
We are going to add more examples for new features where you will need to add more fields if you want to keep that same dataset in the futur. This is not viable long term. It will make examples bigger and bigger.

Also, I'm working on changing some examples in the documentation with a different datasets ( see /meilisearch/documentation#493 ).

Lastly, I don't think a lot of rust users will be trying out every example of the documentation with your code. You had to because you made the code-samples, but users mostly go on the reference page of what the specifically needs and look at the example to understand the usage.